ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -605,4 +605,143 @@ - (NSArray*)serializePacksArray:(NSArray<MGLOfflinePack*>*)packs
+                return;
+            }
+
+            NSArray<id<MGLFeature>> *features;
+            NSArray<NSString *> *styleLayerIdentifiersArray = options[@"layers"];
+            NSSet<NSString *> *styleLayerIdentifiers;
+            if (styleLayerIdentifiersArray) {
+                styleLayerIdentifiers = [NSSet setWithArray:styleLayerIdentifiersArray];
+            }
+
+            if (pointDict) {
+                NSNumber *screenCoordX = pointDict[@"screenCoordX"];
+                NSNumber *screenCoordY = pointDict[@"screenCoordY"];
+                CGPoint point = CGPointMake(screenCoordX.floatValue, screenCoordY.floatValue);
+                features = [mapView visibleFeaturesAtPoint:point inStyleLayersWithIdentifiers:styleLayerIdentifiers];
+            } else {
+                NSNumber *left = rectDict[@"left"];
+                NSNumber *top = rectDict[@"top"];
+                NSNumber *right = rectDict[@"right"];
+                NSNumber *bottom = rectDict[@"bottom"];
+                CGFloat width = right.floatValue - left.floatValue;
+                CGFloat height = bottom.floatValue - top.floatValue;
+                CGRect rect = CGRectMake(left.floatValue, top.floatValue, width, height);
+                features = [mapView visibleFeaturesInRect:rect inStyleLayersWithIdentifiers:styleLayerIdentifiers];
+            }
+
+            NSMutableArray *geoJSONFeatures = [[NSMutableArray alloc] init];
+            for (id <MGLFeature> feature in features) {
+                NSDictionary *geoJSONGeometry = [self geoJSONGeometryFromMGLFeature:feature];
+                NSDictionary *geoJSON = @{ @"type": @"Feature",
+                                           @"id": feature.identifier ? feature.identifier : [NSNull null],
+                                           @"properties": feature.attributes,
+                                           @"geometry": geoJSONGeometry };
+                [geoJSONFeatures addObject:geoJSON];
+            }
+
+            resolve(geoJSONFeatures);
+        }
+    }];
+}
+
+- (NSDictionary*)geoJSONGeometryFromMGLFeature:(id <MGLFeature>)feature
+{
+    NSString *geometryType;
+
+    if ([feature isKindOfClass:[MGLShapeCollectionFeature class]]) {
+        geometryType = @"GeometryCollection";
+        MGLShapeCollectionFeature *shapeCollection = (MGLShapeCollectionFeature *) feature;
+        NSMutableArray *geometries = [[NSMutableArray alloc] init];
+        for (MGLShape <MGLFeature> *shape in shapeCollection.shapes) {
+            [geometries addObject:[self geoJSONGeometryFromMGLFeature:shape]];
+        }
+        return @{ @"type": geometryType,
+                  @"geometries": geometries };
+    }
+
+    NSMutableArray *coordinates = [[NSMutableArray alloc] init];
+
+    if ([feature isKindOfClass:[MGLPointFeature class]]) {
+        geometryType = @"Point";
+        coordinates = (NSMutableArray *) @[@(feature.coordinate.longitude), @(feature.coordinate.latitude)];
```

Comment:
How about not creating an NSArray in the first place? I'm thinking replacing line 680 with the following:
```objective-c
coordinates = [[NSMutableArray alloc] initWithArray:@[@(feature.coordinate.longitude), @(feature.coordinate.latitude)]];
```